### PR TITLE
FIX SECURITY SCANS ON SJ DEMO WEBSITE: As Dan, I want the SJ demo website to have the security scans to pass, so that we are confident the app is secure.

### PR DESCRIPTION
**Pull request by paul-mesnilgrente**
[FIX SECURITY SCANS ON SJ DEMO WEBSITE: As Dan, I want the SJ demo website to have the security scans to pass, so that we are confident the app is secure.](https://www.pivotaltracker.com/story/show/170941400)

STORY
=====

**Acceptance Criteria**
- `bundle audit`, `yarn audit` and `brakeman` tests are green
- The CI jobs don't allow failure anymore

**Notes**

- Gem vulnerabilities:
rack

- Brakeman overview:
Controllers: 5
Models: 6
Templates: 28
Errors: 0
Security Warnings: 2

- Node packages: 1 vulnerabilities found 
- Spun out of https://www.pivotaltracker.com/story/show/169559944

WHAT WAS DONE
==============

- Upgraded rails
- Upgraded simple_cov
- Ignored the force_ssl warning from brakeman
- Fixed the link_to warning from brakeman
- Upgraded yarn packages to have a clean yarn audit
- Updated the .travis.yml to remove the allow_failure section

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -58,23 +58,3 @@ jobs:
...
-      script:
-        - gem install bundler-audit
-        - bundle audit update
-        - bundle audit check
-
-    - stage: test
-      name: 'brakeman check'
-      script:
-        - gem install brakeman
-        - brakeman --run-all-checks
```
Comment (paul-mesnilgrente) [inline .travis.yml:80]: These tests are still in `.travis.yml`. Removing these lines just made them required.

File: app/views/records/show.html.haml [hunk 2]
```diff
@@ -3,7 +3,7 @@
 .grid-container.record
   %figure
     = link_to(image_tag(@record.image_url(original: true), alt: @record.title, itemprop: 'image'),
-              @record.source_url)
+              html_escape(@record.source_url))
```
Comment (daveharris) [inline app/views/records/show.html.haml:6]: Why is this required? The `source_url` shouldn't contain HTML should it?
